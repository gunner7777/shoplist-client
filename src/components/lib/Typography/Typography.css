.Typography-Text {
  font-size: 1rem;
  margin: 3px 0; }

.Typography-Title {
  font-size: 1.35rem;
  margin-bottom: 15px; }

.Typography_Header {
  font-size: 1.5rem; }

.Typography_SubTitle {
  font-size: 1.1rem; }

.Typography_NoMargin {
  margin: 0; }

.Typography_TextBold {
  font-weight: 600; }
