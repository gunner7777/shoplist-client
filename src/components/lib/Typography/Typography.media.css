@media screen and (max-width: 504px) {
  .Typography-Title {
    font-size: 1.2rem; }
  .Typography_Header {
    font-size: 1.3rem; } }

@media screen and (max-width: 445px) {
  .Typography_Header {
    font-size: 1.1rem; }
  .Typography_SubTitle {
    font-size: 1rem;
    margin-top: 10px; } }

@media screen and (max-width: 390px) {
  .Typography_Header {
    font-size: 1rem; } }

@media screen and (max-width: 360px) {
  .Typography-Title {
    font-size: 1.1rem; } }
