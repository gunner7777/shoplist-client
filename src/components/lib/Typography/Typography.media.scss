@media screen and (max-width: 504px) {
  .Typography {
    &-Title {
      font-size: 1.2rem;
    }

    &_Header {
      font-size: 1.3rem;
    }
  }
}

@media screen and (max-width: 445px) {
  .Typography {
    &_Header {
      font-size: 1.1rem;
    }
    &_SubTitle {
      font-size: 1rem;
      margin-top: 10px;
    }
  }
}

@media screen and (max-width: 390px) {
  .Typography {
    &_Header {
      font-size: 1rem;
    }
  }
}

@media screen and (max-width: 360px) {
  .Typography {
    &-Title {
      font-size: 1.1rem;
    }
  }
}