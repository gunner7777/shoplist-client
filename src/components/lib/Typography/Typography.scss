.Typography {
  &-Text {
    font-size: 1rem;
    margin: 3px 0;
  }

  &-Title {
    font-size: 1.35rem;
    margin-bottom: 15px;
  }

  &_Header {
    font-size: 1.5rem;
  }

  &_SubTitle {
    font-size: 1.1rem;
  }

  &_NoMargin {
    margin: 0;
  }

  &_TextBold {
    font-weight: 600;
  }
}