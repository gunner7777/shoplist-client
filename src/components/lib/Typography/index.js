import React from 'react';
import PropTypes from 'prop-types';
import './Typography.scss';
import './Typography.media.scss';

/*
props
  variant: title, text
  tag: p, h1, h2, h3, ...
  modClass
*/

const Typography = ({ tag, variant, modClass, children }) => {
  const textTag = tag || 'p';
  const style = [];
  if (variant === 'title') {
    style.push('Typography-Title');
  } else if (variant === 'text') {
    style.push('Typography-Text');
  }

  const s = modClass ? style.push(modClass) : null;

  return React.createElement(textTag, { className: style.join(' ') }, children);
};

Typography.propTypes = {
  variant: PropTypes.string,
  tag: PropTypes.string,
  modClass: PropTypes.string,
  children: PropTypes.node.isRequired,
};

export default Typography;
