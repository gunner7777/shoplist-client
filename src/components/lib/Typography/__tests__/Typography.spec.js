import React from 'react';
import { shallow } from 'enzyme';
import Typography from '../index';

describe('typography tests', () => {
  it('Typography rendering with default props', () => {
    const TypographyComponent = shallow(<Typography />);
    expect(TypographyComponent).toMatchSnapshot();
  });

  it('Typography rendering with custom props', () => {
    const props = {
      tag: 'h3',
      variant: 'title',
      children: 'custom props',
    };

    const TypographyComponent = shallow(<Typography {...props} />);
    expect(TypographyComponent).toMatchSnapshot();
  });
});
