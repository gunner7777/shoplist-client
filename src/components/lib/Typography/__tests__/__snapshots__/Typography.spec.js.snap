// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typography tests Typography rendering with custom props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Typography
    tag="h3"
    variant="title"
  >
    custom props
  </Typography>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "custom props",
      "className": "Typography-Title",
    },
    "ref": null,
    "rendered": "custom props",
    "type": "h3",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "custom props",
        "className": "Typography-Title",
      },
      "ref": null,
      "rendered": "custom props",
      "type": "h3",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`typography tests Typography rendering with default props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Typography
    tag="p"
    variant="text"
  >
    test
  </Typography>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "test",
      "className": "Typography-Text",
    },
    "ref": null,
    "rendered": "test",
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "test",
        "className": "Typography-Text",
      },
      "ref": null,
      "rendered": "test",
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
